File: dotnet/src/Plugins/Core/Interaction/InteractionPlugin.cs
using Agience.Core;
using Microsoft.Extensions.DependencyInjection;
using Microsoft.SemanticKernel;
using System.ComponentModel;

namespace Agience.Plugins.Core.Interaction
{
    public class InteractionPlugin
    {
        private readonly IInteractionService _interaction;
        private readonly IServiceProvider _serviceProvider;

        public InteractionPlugin (IInteractionService interaction, IServiceProvider serviceProvider)
        {
            _interaction = interaction;
            _serviceProvider = serviceProvider;           
        }

        [KernelFunction, Description("Show a message")]
        public async Task Send(
            [Description("The message to show")] string message
            )
        {
            await _interaction.ReceiveFromAgent(_serviceProvider.GetRequiredService<Agent>(), message);
        }

        [KernelFunction, Description("Receive a message")]
        [return: Description("The received message")]
        public async Task<string> Receive()
        {
            return await _interaction.SendToAgent(_serviceProvider.GetRequiredService<Agent>()) ?? string.Empty;
        }

        [KernelFunction, Description("Show a message and collect a response")]
        [return: Description("The response message")]
        public async Task<string> Interact(
            [Description("The message to show")] string message)
        {
            await Send(message);
            return await Receive();
        }
    }
}

File: core/plugins/python/interaction/interaction_plugin.py
from semantic_kernel.functions import kernel_function

from core.agent import Agent
from utils.service_container import ServiceProvider
from plugins.core.interaction.interaction_service_interface import IInteractionService


class InteractionPlugin:
    def __init__(self, interaction_service: IInteractionService, service_provider: ServiceProvider):
        self._interaction = interaction_service
        self._service_provider = service_provider

    @kernel_function(description="Show a message")
    async def send(self, message: str) -> None:
        agent = self._service_provider.get_required_service(Agent)
        await self._interaction.receive_from_agent(agent, message)

    @kernel_function(description="Receive a message")
    async def receive(self) -> str:
        agent = self._service_provider.get_required_service(Agent)
        result = await self._interaction.send_to_agent(agent)
        return result if result is not None else ""

    @kernel_function(description="Show a message and collect a response")
    async def interact(self, message: str) -> str:
        await self.send(message)
        return await self.receive()
